src/serializationStrategies/XpathRangeSelector/xpath.ts

```diff
@@ -181,21 +181,29 @@ export function getFirstByXPath(path: string, offset: number, referenceElement:
   // for element targets, highlight children might be artifically
   // inflating the range offset, fix.
   if (node && isElement(node)) {
-    let search: Node | null = node.childNodes[offset - 1];
+    let search: Node | null = node.childNodes[0];
+    let offsetElementsFound = 0;
+    let modifyOffset = 0;
+
+    while (search && offsetElementsFound < offset) {
+      offsetElementsFound++;
 
-    while (search) {
       if (isTextOrTextHighlight(search)) {
         search = search.nextSibling;
 
         while (isTextOrTextHighlight(search!)) {
```

Comment:
I guess this function will return false if `search` is `null`, but probably we should change this to avoid using `!`
```
  while (search && isTextOrTextHighlight(search)) {
```

Reply:
this code wasn't originally written in typescript and is very brittle so i've avoided any major restructuring just to make the nullchecks happy until i have more extensive test coverage